.gemini/styleguide.md

```diff
@@ -0,0 +1,402 @@
+# Style Guide
+
+## General Principles
+
+APIs added should be:
+- Scalable
+- Robust
+- Backwards compatible
+- Future-proof
+- Interoperable
+- Must work across all backends
+
+---
+
+# Keras API design guidelines
+
+These guidelines are meant to help focus design discussions and help us create d
+elightful developer experiences.
+
+These are meant as guidelines, not rules: each decision should be debated in its
+ own unique context.
+
+Some text remixed from external references:
+
+- [User experience design for APIs](https://blog.keras.io/user-experience-design
+-for-apis.html)
+- [Notes to Myself on Software Engineering](https://medium.com/s/story/notes-to-
+myself-on-software-engineering-c890f16f4e4d)
+
+
+---
+
+## Design end-to-end workflows, not individual functions and classes.
+
+When developing APIs, start by designing end-to-end workflows, and only sketch o
+ut specific function/class signatures at the end.
```

Comment:
wrapping went wrong throughout this file, words are cut all over the place.

Reply:
fixed it now :P